zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java

```diff
@@ -248,8 +256,19 @@ public void shouldApplyInputMappings() {
             VariableRecordValue::getValue,
             VariableRecordValue::getScopeKey)
         .contains(
-            tuple("x", "1", processInstanceKey),
-            tuple("local_x", "2", adHocSubProcessActivated.getKey()));
+            tuple("activateElements", "[\"A\"]", processInstanceKey),
+            tuple("activateElements", "[\"A\",\"B\"]", adHocSubProcessActivated.getKey()));
+
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limit(AD_HOC_SUB_PROCESS_ELEMENT_ID, ProcessInstanceIntent.ELEMENT_ACTIVATED))
+        .extracting(r -> r.getValue().getElementId(), Record::getIntent)
```

Comment:
💭 For the future contributors: may be something like `describedAs` would be useful to explain what order is expected here.